### PR TITLE
Bluetooth: controller: Update supported commands for ISO

### PR DESCRIPTION
**Pull request by wopu-ot**
Set bits for ISO-related supported commands depending on selected
features and adjust compilation guards for the respective handler
functions.

Signed-off-by: Wolfgang Puffitsch <wopu@demant.com>

### CODE REVIEW

**Review by thoh-ot — APPROVED** [review]

File: subsys/bluetooth/controller/hci/hci.c [hunk 1]
```diff
@@ -1939,7 +1968,23 @@ static void le_remove_cig(struct net_buf *buf, struct net_buf **evt)
...
 #endif /* CONFIG_BT_CENTRAL */
 
-#if defined(CONFIG_BT_CTLR_CONN_ISO)
+#if defined(CONFIG_BT_CTLR_ADV_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
+static void le_iso_transmit_test(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_cp_le_iso_transmit_test *cmd = (void *)buf->data;
+	struct bt_hci_rp_le_iso_transmit_test *rp;
+	uint8_t status;
+	uint16_t handle;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/hci/hci.c:1977]:
```suggestion
	uint16_t handle;
	uint8_t status;
```

File: subsys/bluetooth/controller/hci/hci.c [hunk 2]
```diff
@@ -1963,6 +2008,46 @@ static void le_read_iso_tx_sync(struct net_buf *buf, struct net_buf **evt)
...
 }
+#endif /* CONFIG_BT_CTLR_ADV_ISO || CONFIG_BT_CTLR_CONN_ISO */
+
+#if defined(CONFIG_BT_CTLR_SYNC_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
+static void le_iso_receive_test(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_cp_le_iso_receive_test *cmd = (void *)buf->data;
+	struct bt_hci_rp_le_iso_receive_test *rp;
+	uint8_t status;
+	uint16_t handle;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/hci/hci.c:2019]:
```suggestion
	uint16_t handle;
	uint8_t status;
```

File: subsys/bluetooth/controller/hci/hci.c [hunk 3]
```diff
@@ -1963,6 +2008,46 @@ static void le_read_iso_tx_sync(struct net_buf *buf, struct net_buf **evt)
...
+
+static void le_iso_read_test_counters(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_cp_le_read_test_counters *cmd = (void *)buf->data;
+	struct bt_hci_rp_le_read_test_counters *rp;
+	uint8_t status;
+	uint16_t handle;
+	uint32_t received_cnt;
+	uint32_t missed_cnt;
+	uint32_t failed_cnt;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/hci/hci.c:2038]:
```suggestion
	uint32_t received_cnt;
	uint32_t missed_cnt;
	uint32_t failed_cnt;
	uint16_t handle;
	uint8_t status;
```

**Review by cvinayak — COMMENTED** [review]
Reviewed the bitmap, LGTM.

Minor nitpick on the reverse Christmas tree declarations.

**Review by cvinayak — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Comment by wopu-ot** [comment]
Put on DNM because bit for LE ISO Transmit Test is missing and LE Read ISO Link Quality should be optional. I will update accordingly.

**Comment by Thalley** [comment]
> Put on DNM because bit for LE ISO Transmit Test is missing and LE Read ISO Link Quality should be optional. I will update accordingly.

@wopu-ot How about `LE Set CIG Parameters Test`?

**Comment by wopu-ot** [comment]
> @wopu-ot How about `LE Set CIG Parameters Test`?
It's in line 737/738.

**Comment by Thalley** [comment]
> > @wopu-ot How about `LE Set CIG Parameters Test`?
> 
> It's in line 737/738.

Sorry, I wasn't clear: Shouldn't `LE Set CIG Parameters Test` be optional as well?

**Comment by wopu-ot** [comment]
> > > @wopu-ot How about `LE Set CIG Parameters Test`?
> > 
> > 
> > It's in line 737/738.
> 
> Sorry, I wasn't clear: Shouldn't `LE Set CIG Parameters Test` be optional as well?

That command is not an optional command, it is mandatory if the CIS central role is supported.

BT Core Spec Version 5.3, Vol. 4, Part E, Table 3.1:
> LE Set CIG Parameters Test command ... C39
> ...
> C39: Mandatory if LE Feature (Connected Isochronous Stream - Central) is supported,
otherwise excluded.

**Comment by Thalley** [comment]
> > > > @wopu-ot How about `LE Set CIG Parameters Test`?
> > > 
> > > 
> > > It's in line 737/738.
> > 
> > 
> > Sorry, I wasn't clear: Shouldn't `LE Set CIG Parameters Test` be optional as well?
> 
> That command is not an optional command, it is mandatory if the CIS central role is supported.
> 
> BT Core Spec Version 5.3, Vol. 4, Part E, Table 3.1:
> 
> > LE Set CIG Parameters Test command ... C39
> > ...
> > C39: Mandatory if LE Feature (Connected Isochronous Stream - Central) is supported,
> > otherwise excluded.

I see, thanks. To me it's curious that you can't disable support for this command in production builds, but I suppose there's some reason for that.

**Review by cvinayak — APPROVED** [review]

**Pull request merged**